Dockerfile

```diff
@@ -194,12 +194,14 @@ RUN curl -fsSL -o /final/bin/nextclade2 https://github.com/nextstrain/nextclade/
  && ln -sv nextclade2 /final/bin/nextclade
 
 # Auspice
-# Install Node deps, build Auspice, and link it into the global search path.  A
-# fresh install is only ~40 seconds, so we're not worrying about caching these
-# as we did the Python deps.  Building auspice means we can run it without
-# hot-reloading, which is time-consuming and generally unnecessary in the
-# container image.  Linking is equivalent to an editable Python install and
-# used for the same reasons described above.
+# Building auspice means we can run it without hot-reloading, which is
+# time-consuming and generally unnecessary in the container image.
+# Linking is used so we can overlay the auspice version in the image with
+# --volume=.../auspice:/nextstrain/auspice and still have it globally accessible
+# and importable.
+# Note: even if the version of NPM has platform differences between where
+# Auspice is installed and where it is used (the final image), that does not
+# matter since JavaScript files are not platform-specific.
```

Comment:
Oh, splitting the comment away from c3fbbd9bbe09edb6c14b5cf3c1ad8c56ee2a9aa9 confused me. I've combined the two in 7e7bfb29149e991c74c30c6c7e22e37a9db84fc2 and reworded the comment. I still don't fully understand the implications of `npm install auspice` with a `linux/amd64` image and copying those files onto the `linux/arm64` image, so I'll test more on my M1 Mac.

Reply:
> I still don't fully understand the implications of `npm install auspice` with a `linux/amd64` image and copying those files onto the `linux/arm64` image, so I'll test more on my M1 Mac.

I would think (maybe naively) that at worst this could result in Node.js extensions running under emulation. But, the `--arch` and `--target_arch` flags presumably avoid that?

So (empirically-driven understanding)++

Reply:
That's also the worst case I can think of. The command runs without errors, so at least *something* must have installed. A summary of my understanding:

1. The flags are passed down from `npm` to any precompiling tools that are used.
2. It's up to each dependency to support cross-platform installs through one of those methods (see all the PRs/issues referencing https://github.com/nodejs/node-gyp/issues/829).
3. If a dependency does not support installing for the target platform through `node-gyp`, the calling NPM command will error. I confirmed this by running `npm install --arch=bogus electron` which gives a 404 error.
4. If a dependency does not support installing for the target platform through one of the other precompiling tools, I'm not sure what happens but I assume either:
    1. The calling `npm` command will error, as is the case with `node-gyp`.
    2. Fallback to the host architecture.

Looking at the bigger picture, I think it's fine to use emulation for Auspice on `linux/arm64` hosts.

Reply:
I'm tempted to split this out into a separate PR since `--arch` and `--target_arch` don't seem to be used for any dependencies currently. I can't find a good method to tell if that's the case, but worst case Auspice runs with emulation on `linux/arm64` hosts.

My plan:

1. Replace 7e7bfb29149e991c74c30c6c7e22e37a9db84fc2 with a comment on potential emulation of dependencies.
2. Wait for CI to build a new test image.
3. Verify that Auspice works locally on my M1 machine using the test image.

Reply:
Sounds fine if you want to split it out, but I also think ok to leave it here.

Reply:
I came up with a search pattern for platform-specific dependencies, and did not find anything that sticks out as impactful (`node-notifier` and `puppeteer` would only be used for testing purposes). I'll follow through with the plan.

```
nextstrain:~/auspice $ pwd
/nextstrain/auspice

nextstrain:~/auspice $ find node_modules -executable -type f -exec file {} + | grep 64
node_modules/node-notifier/vendor/mac.noindex/terminal-notifier.app/Contents/MacOS/terminal-notifier:                        Mach-O 64-bit x86_64 executable, flags:<NOUNDEFS|DYLDLINK|TWOLEVEL|PIE>
node_modules/node-notifier/vendor/notifu/notifu64.exe:                                                                       PE32+ executable (GUI) x86-64, for MS Windows
node_modules/puppeteer/.local-chromium/linux-722234/chrome-linux/nacl_irt_x86_64.nexe:                                       ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, BuildID[sha1]=7aeb4f45ea5cec7d8e4184264ad39f0f77bcaee2, stripped
node_modules/puppeteer/.local-chromium/linux-722234/chrome-linux/nacl_helper_bootstrap:                                      ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, BuildID[sha1]=5c5f3935c8f8a15ba325f0d73dfa585fa9390cf9, not stripped
node_modules/puppeteer/.local-chromium/linux-722234/chrome-linux/ClearKeyCdm/_platform_specific/linux_x64/libclearkeycdm.so: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, not stripped
node_modules/puppeteer/.local-chromium/linux-722234/chrome-linux/libEGL.so:                                                  ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, not stripped
node_modules/puppeteer/.local-chromium/linux-722234/chrome-linux/crashpad_handler:                                           ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, not stripped
node_modules/puppeteer/.local-chromium/linux-722234/chrome-linux/chrome_sandbox:                                             ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, not stripped
node_modules/puppeteer/.local-chromium/linux-722234/chrome-linux/chrome:                                                     ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, not stripped
node_modules/puppeteer/.local-chromium/linux-722234/chrome-linux/libGLESv2.so:                                               ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, not stripped
node_modules/puppeteer/.local-chromium/linux-722234/chrome-linux/swiftshader/libEGL.so:                                      ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, not stripped
node_modules/puppeteer/.local-chromium/linux-722234/chrome-linux/swiftshader/libGLESv2.so:                                   ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, not stripped
node_modules/puppeteer/.local-chromium/linux-722234/chrome-linux/nacl_helper:                                                ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, not stripped
```

Reply:
Oh jeez, we're shipping dev deps in the container?! Including a copy of Chromium! I guess we are since NPM installs them by default. Oy. We should (separate from this PR) be using `--omit dev` or equivalent. Looks like that'll roughly reduce uncompressed image size by 600MB.

Reply:
Agreed! Glad the extra digging here was worth it.

Reply:
Replaced 7e7bfb29149e991c74c30c6c7e22e37a9db84fc2 with deabcc1a6fa34a75a0b9e4b6b8440ca04e71962d.

Reply:
https://github.com/nextstrain/docker-base/pull/165

Reply:
Auspice works on my M1 machine using the latest CI image:

```
nextstrain view --docker --image nextstrain/base:branch-victorlin-arm64-cross-compilation zika-tutorial/auspice/
```

Merging!